quinn-proto/src/connection.rs

```diff
@@ -1228,14 +1214,17 @@ impl Connection {
                     // Forget about unacknowledged handshake packets
                     self.handshake_cleanup(&ctx.config);
                 }
-                let closed = match self.process_payload(ctx, now, number, packet.payload.into()) {
-                    Ok(closed) => closed,
-                    Err(e) => {
-                        ctx.events
-                            .push_back((self.handle, Event::ConnectionLost { reason: e.into() }));
-                        return Err(e.into());
-                    }
-                };
+                let closed =
+                    match self.process_payload(ctx, now, number.unwrap(), packet.payload.into()) {
```

Comment:
I think we can just propagate the error here? In fact, I think that was already on master?